pkg/k8splan/watcher.go

```diff
@@ -386,7 +386,9 @@ func validateKC(ctx context.Context, config *rest.Config) error {
 	}
 
 	config.Transport = utilnet.SetTransportDefaults(&http.Transport{
-		DialTLS: func(network, addr string) (net.Conn, error) {
+		Proxy:           http.ProxyFromEnvironment,
+		TLSClientConfig: tlsConfig,
```

Comment:
Per the [documentation](https://pkg.go.dev/net/http#Transport) `TLSClientConfig` will be ignored since we've defined `DialTLSContext`:
```
        // If DialTLSContext is set, the Dial and DialContext hooks are not used for HTTPS
	// requests and the TLSClientConfig and TLSHandshakeTimeout
	// are ignored. The returned net.Conn is assumed to already be
	// past the TLS handshake.
```

Reply:
Fixed